core/circuitbreaker/rule_manager.go

```diff
@@ -245,6 +233,104 @@ func onRuleUpdate(rules []Rule) (err error) {
 	return nil
 }
 
+func AppendRule(r Rule) error {
+	res := r.ResourceName()
+	oldResCbs := breakers[res]
+	if oldResCbs == nil {
+		oldResCbs = make([]CircuitBreaker, 0, 0)
+	}
+	updateMux.Lock()
+	defer func() {
+		updateMux.Unlock()
+		if r := recover(); r != nil {
+			return
+		}
+	}()
+	equalIdx, reuseStatIdx := calculateReuseIndexFor(r, oldResCbs)
```

Comment:
For appending rule, from my perspective, the statistic structure shouldn't be reused.